src/bandit-evaluator.spec.ts

```diff
@@ -0,0 +1,300 @@
+            },
+          ],
+        },
+        action2: {
+          actionKey: 'action2',
+          intercept: 0.3,
+          subjectNumericCoefficients: [
+            { attributeKey: 'age', coefficient: 0.2, missingValueCoefficient: 0.3 },
+          ],
+          subjectCategoricalCoefficients: [
+            {
+              attributeKey: 'color',
+              missingValueCoefficient: 0.6,
+              valueCoefficients: { red: 0.4 },
+            },
+          ],
+          actionNumericCoefficients: [
+            { attributeKey: 'price', coefficient: -0.1, missingValueCoefficient: 0.0 },
+          ],
+          actionCategoricalCoefficients: [
+            {
+              attributeKey: 'category',
+              missingValueCoefficient: -0.2,
+              valueCoefficients: { B: 0.4 },
+            },
+          ],
+        },
+      },
+    };
+
+    it('scores each action', () => {
+      const subjectAttributes: Attributes = { age: 30, location: 'US' };
+      const actions: Record<string, Attributes> = {
+        action1: { price: 25, category: 'A' },
+        action2: { price: 50, category: 'B' },
+        action99: { price: 100, category: 'C' },
+      };
+      const actionScores = exposedEvaluator.scoreActions(subjectAttributes, actions, modelData);
+      expect(Object.keys(actionScores)).toHaveLength(3);
+      expect(actionScores.action1).toBe(0.5 + 30 * 0.1 + 0.2 + 25 * 0.05 + 0.3);
+      expect(actionScores.action2).toBe(0.3 + 30 * 0.2 + 0.6 + -0.1 * 50 + 0.4);
+      expect(actionScores.action99).toBe(1.23); // Default score
+    });
+  });
+
+  describe('weighActions', () => {
+    it('handles no actions', () => {
+      const actionWeights = exposedEvaluator.weighActions({}, 1, 0.1);
+      expect(actionWeights).toStrictEqual({});
+    });
+
+    it('weights a single action at 100%', () => {
+      const scoredActions = { action: 1.23 };
+      const actionWeights = exposedEvaluator.weighActions(scoredActions, 1, 0.1);
+      expect(actionWeights).toStrictEqual({
+        action: 1.0,
+      });
+    });
+
+    it('weighs multiple actions with the same scores', () => {
+      const scoredActions = { action1: 5, action2: 5, action3: 5 };
+      const actionWeights = exposedEvaluator.weighActions(scoredActions, 1, 0.1);
+      expect(Object.keys(actionWeights)).toHaveLength(3);
+      expect(actionWeights.action1).toBeCloseTo(0.33333);
+      expect(actionWeights.action2).toBeCloseTo(0.33333);
+      expect(actionWeights.action3).toBeCloseTo(0.33333);
+    });
+
+    it('weighs multiple actions with different scores', () => {
+      const scoredActions = { action1: 1, action2: 0.5 };
+      const gamma = 10;
+      const actionWeights = exposedEvaluator.weighActions(scoredActions, gamma, 0.1);
+      expect(Object.keys(actionWeights)).toHaveLength(2);
+      expect(actionWeights.action1).toBeCloseTo(0.85714);
+      expect(actionWeights.action2).toBeCloseTo(0.14286);
```

Comment:
At this point, it's probably too much / too confusing to "show our work" as we did for previous tests and just use trusted, manually-computed values.